File: Data Structures & Algorithms/climbing-stairs/submission-1.py
class Solution:
    def climbStairs(self, n: int) -> int:
        def dp(i: int) -> int:
            if i == n:
                return 1
            if i > n:
                return 0
            return dp(i+1) + dp(i+2)
            
        return dp(0)

File: Data Structures & Algorithms/anagram-groups/submission-1.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        let anagrams = {}
        for (let s of strs) {
            let key = s.split('').sort().join('')
            if (key in anagrams) {
                anagrams[key].push(s)
            } else {
                anagrams[key] = [s]
            }
        }
        return Object.values(anagrams)
    }
}
